src/metrics/metric_manager.py

```diff
@@ -0,0 +1,210 @@
+
+        Args:
+            params: All metric params from config file.
+            phase: Current phase Enum.
+
+        Returns:
+            metrics: Metric list as nn.ModuleList for current phase. 
+        """
+        # create added_metric_names set
+        added_log_names = set()
+        metrics = []
+        for metric_params in params:
+            if phase not in metric_params.phases:
+                continue
+            metric = METRICS.get(metric_params.name)(**metric_params.params)
+            mapping = metric_params.mapping
+            log_name = metric_params.log_name if metric_params.log_name is not None else metric_params.name
+            if log_name in added_log_names:
+                mapping_values = list(mapping.values())
+                log_name += + '_' + mapping_values[0] + '_' + mapping_values[1]
+            else:
+                added_log_names.add(log_name)
+
+            metrics.append(MetricWithUtils(metric=metric, mapping=mapping, log_name=log_name))
+
+        metrics = nn.ModuleList(metrics)
+        return metrics
+
+    def forward(self, phase: Phase, *args, **kwargs):
+        """Update states of all metrics on phase loop.
+
+        Args:
+            phase: Phase Enum.
+        """
+        args = list(args)
+        if phase not in self.phases:
+            raise ValueError(f'Incorrect epoch setting. '
+                             f'Please choose one of enum value {self.phases}')
+
+        for metric_with_utils in self.__phase2metrics[phase]:
+            targeted_kwargs = self.map_arguments(metric_with_utils.mapping, kwargs)
+            if targeted_kwargs:
+                # may be we need only update because forward use compute and sync all the processses
+                metric_with_utils(*args, **targeted_kwargs)
+            
+
+    def on_epoch_end(self, phase: Phase) -> Dict[str, Tensor]:
+        """Summarize epoch values and return log.
+        
+        Args:
+            phase: Run metric phase.
+
+        Returns:
+            log: Logging dictionary, there the key is phase/metric_name and value is metric value on phase.
+
+        Raises:
+            ValueError: If phase not in self.phases.
+            ValueError: If metric.compute() return tensor with non zero shape.
+        """
+        if phase not in self.phases:
+            raise ValueError(f'Incorrect epoch setting. '
+                             f'Please choose one of enum value {self.phases}')
+            
+        log = {}
+        for metric_with_utils in self.__phase2metrics[phase]:
+            metric_value = metric_with_utils.compute()
+            if isinstance(metric_value, dict):
+                metric_value = list(metric_value.values())[0]
+            
+            if len(metric_value.shape) != 0:
+                raise ValueError(f'{metric_with_utils.log_name} must compute float value, '
+                                f'not torch tensor with shap {metric_value.shape}.')
```

Comment:
Ok, now we must decide what is good type, i think it's just a number, because we return logs with string value. And some metrics such like Confusion Matrix would be bad to write as string in PL logger. 
What do you think?